opnieuw/retries.py

```diff
@@ -68,98 +68,84 @@ def calculate_exponential_multiplier(
     return multiplier
 
 
-class DoCall:
+class BackoffCalculator:
     """
-    An instance which tells the retry decorator to attempt the function.
-    """
-
-
-class DoWait(NamedTuple):
-    """
-    An instance which tells the retry decorator to wait.
+    Class responsible for calculating backoff periods.
 
-    This is used to calculate the sleep time in seconds for in the retry decorator.
-    """
-
-    # Number of calls made so far.
-    attempts_so_far: int
-    # Wait at least this much.
-    min_seconds: float
-    # Wait at most this much.
-    max_seconds: float
-    # Number of seconds left before the user's requested seconds are exceeded
-    seconds_left: float
-
-
-Action = Union[DoCall, DoWait]
-
-
-class RetryState:
-    """
-    Calculate whether to do the function call or to sleep for retrying.
+    Will consider the maximum amount of backoffs and a maximum backoff window.
     """
 
     def __init__(
         self,
         clock: Clock,
-        max_calls_total: int,
-        retry_window_after_first_call_in_seconds: int,
+        max_backoffs_total: int,
```

Comment:
I think this name is a bit misleading, as the maximum number of backoffs is always 1 less than the `max_calls_total`. I do not see the reason to to change these names, so keeping the `max_calls_total` here seems better.